modules/fs2/src/main/scala/pureconfig/module/fs2/package.scala

```diff
@@ -1,55 +1,37 @@
 package pureconfig.module
 
-import java.io._
+import java.nio.charset.StandardCharsets.UTF_8
 
-import scala.concurrent.ExecutionContext
 import scala.language.higherKinds
 import scala.reflect.ClassTag
-
-import _root_.fs2.{ Stream, io, text }
-import cats.effect.{ Concurrent, Sync, ContextShift }
+import _root_.fs2.{ Stream, text }
+import cats.effect.Sync
 import cats.implicits._
 import com.typesafe.config.{ ConfigFactory, ConfigRenderOptions }
 import pureconfig.{ ConfigReader, ConfigWriter, Derivation }
 import pureconfig.error.ConfigReaderException
 
 package object fs2 {
 
-  private def parseStream[F[_], A](inputStream: InputStream)(implicit F: Sync[F], reader: Derivation[ConfigReader[A]], ct: ClassTag[A]): F[A] = {
-    val streamReader = new InputStreamReader(inputStream)
-    val config = F.delay { ConfigFactory.parseReader(streamReader) }
-    config.flatMap { c =>
-      val loadedConfig = pureconfig.loadConfig(c).leftMap(ConfigReaderException[A])
-      F.fromEither(loadedConfig)
-    }
-  }
-
-  private def pipedStreams[F[_]](implicit F: Sync[F]): F[(InputStream, OutputStream)] = F.delay {
-    val output = new PipedOutputStream()
-    val input = new PipedInputStream(output)
-    (input, output)
-  }
-
   /**
    * Load a configuration of type `A` from the given byte stream.
    *
    * @param configStream a stream of bytes representing the contents of a configuration file
    * @return The returned action will complete with `A` if it is possible to create an instance of type
    *         `A` from the configuration stream, or fail with a ConfigReaderException which in turn contains
    *         details on why it isn't possible
+   *         It can also raise any exception that the stream can raise, as well as any exception that
+   *         ConfigFactory.parseString might throw.
```

Comment:
This isn't very nice, since PureConfig entrypoints handle malformed configs gracefully. You may want to take a look at [ConfigFactoryWrapper](https://github.com/pureconfig/pureconfig/blob/master/core/src/main/scala/pureconfig/backend/ConfigFactoryWrapper.scala), the wrapper we use to call `ConfigFactory` methods in a safe way. Feel free to add a `parseString` method there and use it in this module.

Reply:
Thanks, looks good! Can you just add tests for both parsing errors and `loadConfig` errors, so that this behavior is not accidentally changed in future PRs?

Reply:
Short answer: no, I won't waste my time writing tests for behaviours I didn't change.

I wrote a new, simpler implementation because I like simple code. Take it or leave it…

Reply:
Sure. Thanks anyway.